### ISSUE TITLE
`selectQueryBuilder.limit(0)` is ignored

### ISSUE THREAD

**Issue by alumni** [typeorm/typeorm#7849]
## Issue Description

`qb.limit(0)` has no effect.

### Expected Behavior

The result of `qb.getQuery()` should contain `LIMIT 0`.

### Actual Behavior

The limit is ignored.

### Steps to Reproduce

```ts
connection.createQueryBuilder().from('DUMMY').limit(0).getQuery();
```
Expected:
```sql
SELECT * FROM DUMMY LIMIT 0
```
Actual:
```sql
SELECT * FROM DUMMY
```

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| TypeORM version     | v0.2.34 | 

### Additional Context

`SelectQueryBuilder#createLimitOffsetExpression` checks if `offset` and `limit` are defined like this:

```ts
if (limit && offset) ...
if (limit) ...
if (offset) ...
```

However if the limit is intentionally 0, this check won't work. Instead, it should be checked if the limit is a finite number or, more loosely, if it is defined (not null, not undefined).

### Relevant Database Driver(s)

- [ ] `aurora-data-api`
- [ ] `aurora-data-api-pg`
- [ ] `better-sqlite3`
- [ ] `cockroachdb`
- [ ] `cordova`
- [ ] `expo`
- [ ] `mongodb`
- [x] `mysql`
- [ ] `nativescript`
- [x] `oracle`
- [x] `postgres`
- [ ] `react-native`
- [x] `sap`
- [ ] `sqlite`
- [ ] `sqlite-abstract`
- [ ] `sqljs`
- [x] `sqlserver`

### Are you willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time, and I know how to start.
- [x] Yes, I have the time, but I don't know how to start. I would need guidance. [=> tests]
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by ranjan-purbey** [comment]
The `if` blocks in [createLimitOffsetExpression](https://github.com/typeorm/typeorm/blob/dffd12515c8366b3c33f9c0e2a99c9d60281017b/src/query-builder/SelectQueryBuilder.ts#L1607) should probably check `typeof limit === "numbder"`

**Comment by ranjan-purbey** [comment]
Yes this bug can affect unsuspecting users and imho should be fixed. I can submit a PR if needed.

**Comment by gioboa** [comment]
Thanks @ranjan-purbey feel free to drop a PR for this. 🚀

**Comment by ranjan-purbey** [comment]
This is fixed in https://github.com/typeorm/typeorm/pull/11507 and can be closed @gioboa

**Comment by gioboa** [comment]
Thanks @ranjan-purbey 👍
